asf_installer_survey/notebooks/section_6_analysis.py

```diff
@@ -0,0 +1,1670 @@
+# -*- coding: utf-8 -*-
+# ---
+# jupyter:
+#   jupytext:
+#     cell_metadata_filter: -all
+#     comment_magics: true
+#     custom_cell_magics: kql
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.11.2
+#   kernelspec:
+#     display_name: installersurvey
+#     language: python
+#     name: python3
+# ---
+
+# %% [markdown]
+# ## Section 6: Maintenance and servicing
+# **Aims:** Generating outputs for Section 6 (RQs 6.01-6.25) as planned in [Heat Pump Installer Survey data analysis plan](https://docs.google.com/document/d/1M1nzdf3fyTjipmaKJKViin0EB3e3R1afOQglwMGJaII/edit#heading=h.jzmlm8j0kyve)<br>
+# **Author:** Elysia Lucas<br>
+# **Date:** 2024-04-04<br>
+# <i>Update:</i> 2024-04-17 (Re-coding of free text responses)
+# <p> 
+# N.B. As of 11.04.2024, only CompanySizeOwner and CompanySizeEmployee outputs with version 2 categorisation are generated.
+
+# %% [markdown]
+# ### Importing and pre-processing
+
+# %%
+with open("data_setup.py") as file:
+    exec(file.read())
```

Comment:
I am getting the following error here: `---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
Cell In[1], [line 2](vscode-notebook-cell:?execution_count=1&line=2)
      [1](vscode-notebook-cell:?execution_count=1&line=1) with open("data_setup.py") as file:
----> [2](vscode-notebook-cell:?execution_count=1&line=2)     exec(file.read())
      [4](vscode-notebook-cell:?execution_count=1&line=4) with open("free_text_recode.py") as file:
      [5](vscode-notebook-cell:?execution_count=1&line=5)     exec(file.read())

File <string>:58

File [~/miniconda3/envs/asf_installer_survey/lib/python3.12/site-packages/pandas/io/parquet.py:667](https://file+.vscode-resource.vscode-cdn.net/Users/aidan.kelly/nesta/ASF/asf_installer_survey/asf_installer_survey/notebooks/~/miniconda3/envs/asf_installer_survey/lib/python3.12/site-packages/pandas/io/parquet.py:667), in read_parquet(path, engine, columns, storage_options, use_nullable_dtypes, dtype_backend, filesystem, filters, **kwargs)
    [664](https://file+.vscode-resource.vscode-cdn.net/Users/aidan.kelly/nesta/ASF/asf_installer_survey/asf_installer_survey/notebooks/~/miniconda3/envs/asf_installer_survey/lib/python3.12/site-packages/pandas/io/parquet.py:664)     use_nullable_dtypes = False
    [665](https://file+.vscode-resource.vscode-cdn.net/Users/aidan.kelly/nesta/ASF/asf_installer_survey/asf_installer_survey/notebooks/~/miniconda3/envs/asf_installer_survey/lib/python3.12/site-packages/pandas/io/parquet.py:665) check_dtype_backend(dtype_backend)
--> [667](https://file+.vscode-resource.vscode-cdn.net/Users/aidan.kelly/nesta/ASF/asf_installer_survey/asf_installer_survey/notebooks/~/miniconda3/envs/asf_installer_survey/lib/python3.12/site-packages/pandas/io/parquet.py:667) return impl.read(
    [668](https://file+.vscode-resource.vscode-cdn.net/Users/aidan.kelly/nesta/ASF/asf_installer_survey/asf_installer_survey/notebooks/~/miniconda3/envs/asf_installer_survey/lib/python3.12/site-packages/pandas/io/parquet.py:668)     path,
    [669](https://file+.vscode-resource.vscode-cdn.net/Users/aidan.kelly/nesta/ASF/asf_installer_survey/asf_installer_survey/notebooks/~/miniconda3/envs/asf_installer_survey/lib/python3.12/site-packages/pandas/io/parquet.py:669)     columns=columns,
    [670](https://file+.vscode-resource.vscode-cdn.net/Users/aidan.kelly/nesta/ASF/asf_installer_survey/asf_installer_survey/notebooks/~/miniconda3/envs/asf_installer_survey/lib/python3.12/site-packages/pandas/io/parquet.py:670)     filters=filters,
    [671](https://file+.vscode-resource.vscode-cdn.net/Users/aidan.kelly/nesta/ASF/asf_installer_survey/asf_installer_survey/notebooks/~/miniconda3/envs/asf_installer_survey/lib/python3.12/site-packages/pandas/io/parquet.py:671)     storage_options=storage_options,
    [672](https://file+.vscode-resource.vscode-cdn.net/Users/aidan.kelly/nesta/ASF/asf_installer_survey/asf_installer_survey/notebooks/~/miniconda3/envs/asf_installer_survey/lib/python3.12/site-packages/pandas/io/parquet.py:672)     use_nullable_dtypes=use_nullable_dtypes,
    [673](https://file+.vscode-resource.vscode-cdn.net/Users/aidan.kelly/nesta/ASF/asf_installer_survey/asf_installer_survey/notebooks/~/miniconda3/envs/asf_installer_survey/lib/python3.12/site-packages/pandas/io/parquet.py:673)     dtype_backend=dtype_backend,
    [674](https://file+.vscode-resource.vscode-cdn.net/Users/aidan.kelly/nesta/ASF/asf_installer_survey/asf_installer_survey/notebooks/~/miniconda3/envs/asf_installer_survey/lib/python3.12/site-packages/pandas/io/parquet.py:674)     filesystem=filesystem,
    [675](https://file+.vscode-resource.vscode-cdn.net/Users/aidan.kelly/nesta/ASF/asf_installer_survey/asf_installer_survey/notebooks/~/miniconda3/envs/asf_installer_survey/lib/python3.12/site-packages/pandas/io/parquet.py:675)     **kwargs,
    [676](https://file+.vscode-resource.vscode-cdn.net/Users/aidan.kelly/nesta/ASF/asf_installer_survey/asf_installer_survey/notebooks/~/miniconda3/envs/asf_installer_survey/lib/python3.12/site-packages/pandas/io/parquet.py:676) )

File [~/miniconda3/envs/asf_installer_survey/lib/python3.12/site-packages/pandas/io/parquet.py:267](https://file+.vscode-resource.vscode-cdn.net/Users/aidan.kelly/nesta/ASF/asf_installer_survey/asf_installer_survey/notebooks/~/miniconda3/envs/asf_installer_survey/lib/python3.12/site-packages/pandas/io/parquet.py:267), in PyArrowImpl.read(self, path, columns, filters, use_nullable_dtypes, dtype_backend, storage_options, filesystem, **kwargs)
...
--> [882](https://file+.vscode-resource.vscode-cdn.net/Users/aidan.kelly/nesta/ASF/asf_installer_survey/asf_installer_survey/notebooks/~/miniconda3/envs/asf_installer_survey/lib/python3.12/site-packages/pandas/io/common.py:882)         handle = open(handle, ioargs.mode)
    [883](https://file+.vscode-resource.vscode-cdn.net/Users/aidan.kelly/nesta/ASF/asf_installer_survey/asf_installer_survey/notebooks/~/miniconda3/envs/asf_installer_survey/lib/python3.12/site-packages/pandas/io/common.py:883)     handles.append(handle)
    [885](https://file+.vscode-resource.vscode-cdn.net/Users/aidan.kelly/nesta/ASF/asf_installer_survey/asf_installer_survey/notebooks/~/miniconda3/envs/asf_installer_survey/lib/python3.12/site-packages/pandas/io/common.py:885) # Convert BytesIO or file objects passed with an encoding

FileNotFoundError: [Errno 2] No such file or directory: '/mnt/g/Shared drives/A Sustainable Future/1. Reducing household emissions/2. Projects Research Work/36. Installer survey/05 survey data/20240201_Installer_survey_clean_data_anonymised.parquet'`. I am going to download and read locally for the moment I guess I may have to have google drive downloaded as an application for this to be mounted on my system?

Reply:
Ah yes sorry - I have the shared Google drive as a network drive on my PC (which is then mounted in WSL). Please let me know if you have any trouble finding it in the shared drive to download